Add celltyping to external instructions

sjspielman opened the pull request:
Closes #499 

This PR adds external cell type documentation. There is a decent amount here, so I expect >1 rounds of review! (reminder that I am out this Thursday+Friday, which may affect when reviewers want to schedule their time to review this!).

I have added documentation throughout for adding submitter annotations and for performing cell typing, including an example celltype metadata file. Any organizational or wording feedback let me know! There is also one `TODO` in there about whether we want to say "want a different organ for cellassign? let us know," so please make sure to comment there.

Worth noting that the internal docs are still under review (#500) and this PR is not stacked; we may want to add more relative links between these docs files, but I'd like to wait until #500 is all set first. Any more relative links can be added later in this PR or in a subsequent PR.

While working on these docs, I had some thoughts about how we have set up the cell type annotation metadata file - specifically, do _actually_ want this file to include the columns `singler_ref_file` and `cellassign_ref_file`? You'll see the docs are a bit weird (although I think fairly clear) for the content of these columns. We may wish to alter the cell type process to only accept singler & cell assign reference _names_, and we would create the meta variables for the reference paths themselves in the celltype annotation subworkflow. If we want to take that approach, I'd update docs here accordingly and then in a separate PR we can update how we handle the metadata in the workflow.

jashapiro commented:
I wonder if we want to wait a bit on this PR until we have implemented https://github.com/AlexsLemonade/scpca-nf/issues/470? The reason is that we will be able to better model "good" file names and version tracking at that point.

sjspielman commented:
> I wonder if we want to wait a bit on this PR until we have implemented https://github.com/AlexsLemonade/scpca-nf/issues/470? The reason is that we will be able to better model "good" file names and version tracking at that point.

Yeah, that seems fine! I think we can leave this PR open and I can come back to it later. I don't foresee many, if any, file conflicts.

sjspielman commented:
Places to keep 👀  -

- https://github.com/AlexsLemonade/ScPCA-admin/issues/690
  - "sister" issue https://github.com/AlexsLemonade/scpca-nf/issues/537
- https://github.com/AlexsLemonade/scpca-nf/issues/470

sjspielman commented:
Time to revive this PR! I struggled a little bit with the level of detail to provide _here_ vs provide in https://github.com/AlexsLemonade/scpca-nf/pull/555 (updates to internal instructions; please have a peek there for context!). 
So, as part of leaving a first fresh review, please share any thoughts about how into the weeds we want to get with these docs. I am also not convinced that all of my phrasing is clear, so please flag any spots that make no/little sense! 

Related to what level of detail we want to achieve here - [this comment from before](https://github.com/AlexsLemonade/scpca-nf/pull/502#discussion_r1355751817) remains unaddressed, in part because we changed PanglaoDB references to be `<thing>-compartment` (I resolved the comment to reduce clutter, but it's up there!).

> We define these from the organ groupings in `celltype-reference-metadata.tsv`, but they could be built differently. I think we want to be sure to include the information about that file and how we created it/how it might be used to generate alternative references.

jashapiro commented on external-instructions.md:
```diff
@@ -428,7 +434,79 @@ As an example, the Dockerfile that we used to build Space Ranger can be found [h
 After building the docker image, you will need to push it to a [private docker registry](https://www.docker.com/blog/how-to-use-your-own-registry/) and set `params.SPACERANGER_CONTAINER` to the registry location and image id in the `user_template.config` file.
 *Note: The workflow is currently set up to work only with spatial transcriptomic libraries produced from the [Visium Spatial Gene Expression protocol](https://www.10xgenomics.com/products/spatial-gene-expression) and has not been tested using output from other spatial transcriptomics methods.*
 
+## Cell type annotation
```
I think this whole section probably should be moved above the special considerations section. I might actually move Output files up as well, as I think people probably want to know about outputs for the normal workflow before details for other cases, but harder to say there.

jashapiro reviewed (commented):
Okay, now that we have most of the workflow ready for this, I left some real comments.

The biggest thing I see right now is that the directions about the reference files are probably too specific to the details of how we are using the workflow rather than its capabilities. So I would keep this document focused on the kinds of files needed, the required format of those files (singleR model, TSV file of marker genes, etc.), and the file name format that the scripts expect. We might want to mention what we are using, and where those files are available, but that should not be the focus of these instructions.

We may want to have a separate document about the cell reference construction, but I would actually expect most of that to live in the main docs repository, and we should be able to link to that for details. You have some of those details in the internal docs, but I don't think they are quite right there either, as they include things (like what  does `bone-compartment mean?`) that users will want to know even if they never run the workflow themselves. 

Relatedly (but this will get copied into another issue), we want to continue to be able to accommodate users who do not have access to S3, so we need to update [get_refs.py](https://github.com/AlexsLemonade/scpca-nf/blob/main/get_refs.py) to download the cell type reference files from S3. I think it is probably fine to keep full paths in the project reference file, but we might want to write out the cell type reference paths somewhere as part of that process. Alternatively, we could make the cell type reference path a parameter and make the actual file paths relative.

sjspielman commented on external-instructions.md:
```diff
@@ -428,7 +434,79 @@ As an example, the Dockerfile that we used to build Space Ranger can be found [h
+  - Please consult the [`celldex` documentation](https://bioconductor.org/packages/release/data/experiment/vignettes/celldex/inst/doc/userguide.html) to determine which of these references, if any, is most suitable for your dataset.
+- `CellAssign` annotation uses marker gene set references from [PanglaoDB](https://panglaodb.se/).
+  Organ-based references were compiled by the Data Lab to represent common organ/tissue groupings.
+  Available references include `blood-compartment`, `brain-compartment`, and `muscle-compartment`.
```
@allyhawkins where are the specific organs relevant to each reference documented? And/or where is the code that creates these?

I looked in `references` but then just ended up opening this issue #561.. I don't immediately see anything in `scpca-admin` either?

jashapiro replied:
This file should be referenced, as it has the main details required: https://github.com/AlexsLemonade/scpca-nf/blob/main/references/celltype-reference-metadata.tsv

sjspielman commented on external-instructions.md:
```diff
@@ -428,7 +434,79 @@ As an example, the Dockerfile that we used to build Space Ranger can be found [h
+
+- `SingleR` annotation uses references from the [`celldex` package](https://bioconductor.org/packages/release/data/experiment/html/celldex.html).
+Available reference options include `BlueprintEncodeData`, `DatabaseImmuneCellExpressionData`, `HumanPrimaryCellAtlasData`, and `MonacoImmuneData`.
+  - Available `SingleR` reference files that you can specify in the `celltype_project_metafile` are saved in `s3:://scpca-references/celltype/singler_models` and are formatted as `<singler_ref_name>_celldex_<celldex_version>_model.rds`.
```
> I'd just name the default here

Do you mean the BlueprintEncode file name?

sjspielman commented on external-instructions.md:
```diff
@@ -19,6 +19,11 @@
+  - [Repeating cell type annotation](#repeating-cell-type-annotation)
 - [Output files](#output-files)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
```
yes. yes we do 🌮

sjspielman commented:
A more general reply to several of the comments about references and defaults since I think we need some broader discussion before I dive back in.

While the references we have prepared for SingleR and CellAssign are not _required_, the level of docs needed to explain how to make and name your own reference (e.g. you need a pre-"trained" singler reference with certain additional aspects in the metadata to play smoothly with the rest of the code) is a bigger lift that I think necessitates its own PR.
Furthermore, regarding this comment about the example references file - https://github.com/AlexsLemonade/scpca-nf/pull/502#discussion_r1391793356

> If these need to be full paths, we should include full paths here.

We have a lot of these paths baked in as parameters here: https://github.com/AlexsLemonade/scpca-nf/blob/4983ecd9f2720f714ae0d0e37528f742338508ea/config/reference_paths.config#L13-L20

This means, to provide one's own cell typing references, one needs to either:
- just override `ref_rootdir` but keep all files structured as expected in a nested `celltype` directory
- override a bunch of params to have your own file organization

---

All this is to say - how do we strike a balance here besides explaining use of defaults vs DIY references? Plus, if we also might move celltyping to the "special considerations" section (https://github.com/AlexsLemonade/scpca-nf/pull/502#discussion_r1391891489), we'd have a really hefty set of "special considerations for this special consideration" instructions.

sjspielman commented on config/reference_paths.config:
Our discussion prompted me to look more closely at this file, and then I got confused about what exactly was what, so I took at stab at more high-level comments. I think it was probably me who suggested comments like `// output from train_singler_models() process, and input to classify_singler()`, but now I think referring to workflows, not processes, makes much more sense for clarity.

allyhawkins reviewed (commented):
I added mostly some initial organization comments. One thing I do wonder is if we want to separate some of these parts into different PRs. Specifically, the entire reference section since I think that's less straight forward.

sjspielman commented on external-instructions.md:
```diff
@@ -426,9 +423,80 @@ For licensing reasons, we cannot provide a Docker container with Space Ranger fo
+| `cell_barcode`         | The cell id with the given annotation label     |
+| `cell_type_assignment` | The annotation label for that cell              |
+
+Optionally, you can also include a column `cell_type_ontology` with ontology labels corresponding to the given annotation label.
```
>  then I would say, but wait, if you want to provide your own labels you can do that in addition to or instead.

Fighting the urge to write "But wait, there's more!"

sjspielman commented on external-instructions.md:
```diff
@@ -426,9 +423,80 @@ For licensing reasons, we cannot provide a Docker container with Space Ranger fo
+
+`scpca-nf` can perform cell type annotation using two complementary methods: the reference-based method [`SingleR`](https://bioconductor.org/packages/release/bioc/html/SingleR.html) and the marker-gene based method [`CellAssign`](https://github.com/Irrationone/cellassign).
+
+You can turn on cell type annotation by using the `--perform_celltyping` flag.
```
Noting that given other broad rearrangements, I kept the references section below the the explanation of this metadata file. Let's see how it looks now in the context of other changes, and can still move it if it needs to move!

sjspielman replied:
Just kidding!! References now above the metadata description - https://github.com/AlexsLemonade/scpca-nf/pull/502/commits/5e8aafd68e1106e1b2681770ac3fd02eb4bf8eba

jashapiro commented on examples/README.md:
```diff
@@ -2,37 +2,41 @@
-These files should be used as an example of formats and content, but note that the values in these files may not be applicable or sufficient to allow running `scpca-nf` to be used directly on your system.
+This directory contains the following example files:
+
+- An example [metadata file](../external-instructions.md#prepare-the-run-metadata-file) for the `scpca-nf` workflow.
```
Looking at the directory, we also have a sample metadata file and multiplex pools file. So we probably want to note all of these. And maybe we want to give the file names themselves? 
```suggestion
- An example [run metadata file](../external-instructions.md#prepare-the-run-metadata-file) for the `scpca-nf` workflow.
```

jashapiro replied:
Not sure if you accidentally marked this as resolved?

sjspielman replied:
Ahh, no I intentionally resolved it... but for some reason my brain read your comment and, naturally, made this change (and similar) instead :scream: https://github.com/AlexsLemonade/scpca-nf/blob/2df686b55dddf28008c4ecc45e05db9b18ebf083/external-instructions.md?plain=1#L150-L151

sjspielman replied:
I'm not mad about the changes I made, but they are just not at all what your comment was....

jashapiro commented on external-instructions.md:
```diff
@@ -15,6 +15,13 @@
     - [Additional reference files](#additional-reference-files)
     - [Downloading container images](#downloading-container-images)
 - [Repeating mapping steps](#repeating-mapping-steps)
+- [Cell type annotation](#cell-type-annotation)
```
Let's add a TOC title?

jashapiro commented on external-instructions.md:
```diff
@@ -126,16 +129,17 @@ To run the workflow, you will need to create a tab separated values (TSV) metada
-| `feature_barcode_geom` | A salmon `--read-geometry` layout string. <br> See https://github.com/COMBINE-lab/salmon/releases/tag/v1.4.0 for details (only required for ADT and cellhash samples)                                            |
-| `slide_section`        | The slide section for spatial transcriptomics samples (only required for spatial transcriptomics)                                                                                                                |
-| `slide_serial_number`  | The slide serial number for spatial transcriptomics samples (only required for spatial transcriptomics)                                                                                                          |
+| column_id                   | contents                                                                                                                                                                                                                     |
+| --------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
Because I agree that wide table formatting is annoying.
```suggestion
<!-- prettier-ignore -->
| column_id | contents |
| --------- | -------- |
```

Feel free to remove all the extra spaces below as well!

jashapiro commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+   You will need to provide the path/uri to this file as a workflow parameter, which you can specify either on the command line or define in your configuration file.
+3. Run the workflow with the `--perform_celltyping` flag.
+
+You would then run the workflow with cell type annotation from the command line as, for example,
```
Making phrasing more consistent with previous code chunks? (Notably the colon)
```suggestion
You would then run the workflow with cell type annotation with a  command line such as the following:
```

jashapiro commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+
+When cell typing is turned on with `--perform_celltyping`, `scpca-nf` will skip annotation for any libraries whose cell type annotation results already exist in the `checkpoints` folder of the given library's output directory, and those reference dataset version is unchanged.
+
+This saves substantial processing time and cost if the cell type annotation reference versions are unchanged.
```
(people's perceived costs may not depend on time if they are running on a "free" server)
```suggestion
This saves substantial processing time if the cell type annotation reference versions are unchanged.
```

jashapiro reviewed (commented):
Looks good! I had a few relatively minor comments to add, with the biggest one a thought about how we might want to let people know what the current version of the reference files is. This interacts a bit with #565, where I added files to track this to S3. Whether those files stay as bare text or change to `json` and add a bit more content is not yet decided in my mind... feel free to weigh in there!

allyhawkins commented on external-instructions.md:
````diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+   You will need to provide the path/uri to this file as a workflow parameter, which you can specify either on the command line or define in your configuration file.
+3. Run the workflow with the `--perform_celltyping` flag.
+
+You would then run the workflow with cell type annotation from the command line as, for example,
+
+```sh
+nextflow run AlexsLemonade/scpca-nf \
+  --perform_celltyping \
+  --celltype_project_metafile examples/example_project_celltype_metadata.tsv
+```
````
```suggestion
Once you have followed the above steps and added the path/uri to the `celltype_project_metafile` to your configuration file, you can use the following command to run the workflow with cell type annotation:

```sh
nextflow run AlexsLemonade/scpca-nf \
  --perform_celltyping 
```
```
I want to be explicit that the above steps need to be completed. Also I'm removing defining the path here, but noting that it should be in the config file.

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+
+### Choosing reference datasets
+
+The Data Lab has compiled several references, listed in [`celltype-reference-metadata.tsv`](references/celltype-reference-metadata.tsv) and available from S3, for use in cell type annotation.
```
```suggestion
The Data Lab has compiled several references, listed in [`celltype-reference-metadata.tsv`](references/celltype-reference-metadata.tsv). 
All references listed in this table are publicly available on S3 for use with cell type annotation.
```

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+Available references include `BlueprintEncodeData`, `DatabaseImmuneCellExpressionData`, `HumanPrimaryCellAtlasData`, and `MonacoImmuneData`.
+Please consult the [`celldex` documentation](https://bioconductor.org/packages/release/data/experiment/vignettes/celldex/inst/doc/userguide.html) to determine which of these references, if any, is most suitable for your dataset.
+
+`SingleR` reference files that you can specify in the `celltype_project_metafile` are stored in `s3:://scpca-references/celltype/singler_models` and are formatted as `<singler_ref_name>_celldex_<celldex_version>_model.rds`.
```
A thought I had reading this... do we want to provide a list of currently available ref files and filenames in the repo for users to easily find which reference would be suitable for them? I'm thinking we could write out this file as part of `build-celltype-ref.nf`.

jashapiro replied:
If we want to write out such a file, we would probably want to keep it on S3 still? A JSON file with all of the currently supported references would be pretty nice, and it could actually simplify the code in #565 a bit.

allyhawkins replied:
Is there any reason we couldn't write it out to the repo? And store in `references`? We can provide the output directory and use `${projectDir}` to tell Nextflow where to store it?

allyhawkins replied:
We could take it one step further and create a GHA for running the reference workflow anytime the metadata or workflow gets updated... then that would ensure that file stays up to date with changes. But first we should get the file to be written out.

sjspielman replied:
> A JSON file with all of the currently supported references would be pretty nice, and it could actually simplify the code in https://github.com/AlexsLemonade/scpca-nf/pull/565 a bit.

So far this seems the like move to me, written out directly to S3 by the references workflow. 
I wonder if we could work out the opposite of our `sample_info` S3 syncing; when the JSON file changes in S3, file a PR to `scpca-nf`? I don't exactly know how we'd set that up, though. Maybe something with time stamps..?

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+#### `SingleR` references
+
+By default, `SingleR` annotation uses references from the [`celldex` package](https://bioconductor.org/packages/release/data/experiment/html/celldex.html).
+Available references include `BlueprintEncodeData`, `DatabaseImmuneCellExpressionData`, `HumanPrimaryCellAtlasData`, and `MonacoImmuneData`.
```
We currently don't support these, and don't plan to. But should we just add them when we build references, even if we don't use them right away?

allyhawkins replied:
You are correct! This is why we need a list somewhere other than S3 😂

sjspielman replied:
> This is why we need a list somewhere other than S3 😂

☝️☝️☝️☝️☝️

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+#### `CellAssign` references
+
+By default, `CellAssign` annotation uses marker gene set references from [PanglaoDB](https://panglaodb.se/), as compiled by the Data Lab to represent common organ/tissue groupings.
+Available references include `blood-compartment`, `brain-compartment`, and `muscle-compartment`.
```
We're going to have more available combinations, so I don't like explicitly stating that here. I think you could state these are examples, but I think we should compile a list of all our supported options and the file names for users.

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+### Preparing the cell type project metadata file
+
+All libraries within a given project will use the same reference dataset for each of `SingleR` and `CellAssign`, respectively.
+The `celltype_project_metafile` file should contain these five columns with the following information (see the example file in [`examples/example_project_celltype_metadata.tsv`](examples/example_project_celltype_metadata.tsv)):
```
Can you turn the link to the example project file into a button? Like we have the other example links.

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+
+| column_id             | contents                                                                                                                              |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| `scpca_project_id`    | Project ID matching values in the metadata file                                                                                       |
```
I would link to the section for preparing the metadata file here.

allyhawkins commented on external-instructions.md:
```diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+
+### Repeating cell type annotation
+
+When cell typing is turned on with `--perform_celltyping`, `scpca-nf` will skip annotation for any libraries whose cell type annotation results already exist in the `checkpoints` folder of the given library's output directory, and those reference dataset version is unchanged.
```
Just a note that we are actually checking the files and not the version.

allyhawkins commented on external-instructions.md:
````diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+To force repeating the cell type annotation process, use the `--repeat_celltyping` flag along with the `--perform_celltyping` flag at the command line:
+
+```sh
+nextflow run AlexsLemonade/scpca-nf \
+  --perform_celltyping \
+  --repeat_celltyping \
+  --celltype_project_metafile examples/example_project_celltype_metadata.tsv
````
```suggestion
nextflow run AlexsLemonade/scpca-nf \
  --perform_celltyping \
  --repeat_celltyping 
```

allyhawkins commented on external-instructions.md:
````diff
@@ -340,6 +340,99 @@ nextflow run AlexsLemonade/scpca-nf \
+  --celltype_project_metafile examples/example_project_celltype_metadata.tsv
+```
+
+### Providing existing cell type labels
````
Just noting that I think this is actually sufficient or providing submitter cell types, so you can probably ignore my earlier comment about temporarily removing it. I do think we want to add an example file though.

sjspielman replied:
I realize I misinterpreted your earlier comment! When you said move this to later on, you meant do it later on (#440) but I read it as "move it later in the document"! So....maybe it worked out ok? example file - https://github.com/AlexsLemonade/scpca-nf/pull/502/commits/38dba37ba7d0321d4c046dc0d01bb7e7b3b4f9cb

If not, I can remove all this and stash to tackle it later.

allyhawkins replied:
No I did mean to move it down here, but also that we could tackle it separately. I think what you have now is fine though!

sjspielman commented on external-instructions.md:
```diff
@@ -426,9 +518,7 @@ For licensing reasons, we cannot provide a Docker container with Space Ranger fo
-
+_Note: The workflow is currently set up to work only with spatial transcriptomic libraries produced from the [Visium Spatial Gene Expression protocol](https://www.10xgenomics.com/products/spatial-gene-expression) and has not been tested using output from other spatial transcriptomics methods._
 
 ## Output files
```
> maybe we can have a section of "Workflow options" and throw a few (or at least one) other things in there?

I like this concept, but nothing is really jumping out at me as along belonging in that section. Though, one day I'm sure more things will arise! The only "maybe" is to also move the section on _repeating_ cell typing into it too?

sjspielman commented:
General comment - I think we're going to want to come back to these instructions after we determine the approach for surfacing references to users. Let's focus on flagging areas in this PR where it should happen, but tackle it in a subsequent PR so this one doesn't keep expanding much more.

sjspielman commented:
Ready for another look! Again, let's aim to get this in soon since this PR is getting too big, and we can follow up with a  subsequent PR where necessary. Note that I left some `TODO` text here for where we can point users to available references.
It also seems that I uhh accidentally tackled #440 as well; can revert some of those change here or just push forward!

allyhawkins commented on external-instructions.md:
```diff
@@ -327,108 +330,108 @@ You will also need to set the `singularity.cacheDir` variable to match this loca
-The `feature_barcode_file` itself is a tab separated file with one line per barcode and no header.
-The first column will contain the barcode or antibody ID and the second column the barcode nucleotide sequence.
-For example:
+By default, `SingleR` annotation uses references from the [`celldex` package](https://bioconductor.org/packages/release/data/experiment/html/celldex.html).
```
Can you file an issue to update this section on cell type references, after #565 goes in.

sjspielman replied:
fyi im on this, have a note waiting for it :)

allyhawkins reviewed (approved):
🎉

jashapiro reviewed (commented):
The external instructions look good, but I do want https://github.com/AlexsLemonade/scpca-nf/pull/502/files#r1394368472 completed before we merge this in (or an issue about those changes, but it seems small enough to sneak in here).

sjspielman commented:
Actually updated the `README.md` in https://github.com/AlexsLemonade/scpca-nf/pull/502/commits/e8b857d5665a24cc8ffead333f3378a9564f3aac, but I'm not sure about how I've written it. The other way I might have gone was, for example. What do you think?

```
- `example_run_metadata.tsv`: An example [run metadata file](../external-instructions.md#prepare-the-run-metadata-file) for the `scpca-nf` workflow.
```

jashapiro commented on examples/README.md:
```diff
@@ -2,37 +2,44 @@
-These files should be used as an example of formats and content, but note that the values in these files may not be applicable or sufficient to allow running `scpca-nf` to be used directly on your system.
+This directory contains the following example files:
+
+- An example [Nextflow configuration file (`user_template.config`)](../external-instructions.md#configuration-files) for the `scpca-nf` workflow.
+- An example [`run_metadata.tsv` file](../external-instructions.md#prepare-the-run-metadata-file) for the `scpca-nf` workflow.
+- An example [`sample_metadata.tsv` file](../external-instructions.md#prepare-the-sample-metadata-file) for the `scpca-nf` workflow.
+- An example [`multiplex_pools.tsv` file](../external-instructions.md#multiplexed-cellhash-libraries) for the `scpca-nf` workflow.
+- An example [`project_celltype_metadata.tsv` file](../external-instructions.md#preparing-the-cell-type-project-metadata-file) for performing optional cell type annotation in the `scpca-nf` workflow.
+- An example [submitter cell type annotation file](../external-instructions.md#providing-existing-cell-type-labels) for optionally providing previously-obtained cell type annotations to the `scpca-nf` workflow.
```
I know I said maybe the file names, but if we do that, we should use the actual file names in this directory. And because of that, I'm rethinking that idea... maybe just the description is sufficient for people to infer the name of the file?

jashapiro commented:
> The other way I might have gone was, for example -
> 
> ```
> - `example_run_metadata.tsv`: An example [run metadata file](../external-instructions.md#prepare-the-run-metadata-file) for the `scpca-nf` workflow.
> ```
> 
> What do you think?

Oh, I missed that comment. I like this version!

sjspielman commented:
Updated, but in table form which gives better spacing, now that we're using colons.

I also removed some changes that went in but in retrospect I think were overkill, and also we decided not to do them -  https://github.com/AlexsLemonade/scpca-nf/issues/469#issuecomment-1798673732

There is no longer an example submitter file. I think if we circle back later and decide we want it included in the example output, it should be a separate issue/PR discussion. But, there is still a section that says you could provide a submitter file if you want.

sjspielman commented:
@allyhawkins tagging you back in too if you have more thoughts on submitter example files..sorry for all the back and forth on that!!!

allyhawkins commented:
> @allyhawkins tagging you back in too if you have more thoughts on submitter example files..sorry for all the back and forth on that!!!

Look good to me, I don't have any other suggestions on that front.

jashapiro reviewed (approved):
👍🏼

sjspielman commented:
:+1:
